metricbeat/module/kubernetes/node/node.go

```diff
@@ -89,25 +87,59 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
-func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	m.enricher.Start()
 
 	body, err := m.mod.GetKubeletStats(m.http)
 	if err != nil {
-		return errors.Wrap(err, "error doing HTTP request to fetch 'node' Metricset data")
-
+		m.Logger().Error(err)
+		reporter.Error(err)
+		return
 	}
 
 	event, err := eventMapping(body)
 	if err != nil {
-		return errors.Wrap(err, "error in mapping")
+		m.Logger().Error(err)
+		reporter.Error(err)
+		return
 	}
 
 	m.enricher.Enrich([]common.MapStr{event})
 
-	reporter.Event(mb.TransformMapStrToEvent("kubernetes", event, nil))
+	var moduleFieldsMapStr common.MapStr
+	moduleFields, ok := event[mb.ModuleDataKey]
+	if ok {
+		moduleFieldsMapStr, ok = moduleFields.(common.MapStr)
+		if !ok {
+			m.Logger().Errorf("error trying to convert '%s' from event to common.MapStr", mb.ModuleDataKey)
+		}
+	}
+	delete(event, mb.ModuleDataKey)
 
-	return nil
+	e := mb.Event{
+		MetricSetFields: event,
+		ModuleFields:    moduleFieldsMapStr,
+		Namespace:       "kubernetes.node",
+	}
+
+	// add root-level fields like ECS fields
+	var metaFieldsMapStr common.MapStr
+	metaFields, ok := event["meta"]
+	if ok {
+		metaFieldsMapStr, ok = metaFields.(common.MapStr)
+		if !ok {
+			m.Logger().Errorf("error trying to convert '%s' from event to common.MapStr", "meta")
+		}
+		delete(event, "meta")
+		e.RootFields = metaFieldsMapStr
+	}
```

Comment:
This or similar logic seems to be repeated along all metricsets, could this be moved to some common helper?